### ISSUE TITLE
[Legitimate Site Blocked] cactus.store

### ISSUE THREAD

**Issue by tmartinez88** [MetaMask/eth-phishing-detect#6397]
https://cactus.store/

**Comment by devenblake** [comment]
Duplicate of #6368 but should have been fixed with #6325. Can you send a screenshot of the block notice?

**Comment by deshvin** [comment]
No response, not getting blocked on my end, can be re-opened if necessary.
